src/api/lib/data.ts

```diff
@@ -1,8 +1,10 @@
 import fs from 'fs'
 import { DB, ServiceKey } from '../types/admin'
+import { Mapping } from '../types/general'
 
 const data: DB = {
-  serviceKeys: []
+  serviceKeys: [],
+  mappings: []
```

Comment:
when the service restarts, your `mappings` value will always reset to `[]`. Then, you are not saving anything right? Look a few lines below, you will see `fs.readFile`. There, you need to do `data.mappings = fileData....`